proofs/tree.go

```diff
@@ -43,35 +43,39 @@ import (
 
 	"regexp"
 
-	"github.com/centrifuge/go-merkle"
 	"github.com/centrifuge/precise-proofs/proofs/proto"
 	"github.com/golang/protobuf/descriptor"
 	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/timestamp"
 	"github.com/iancoleman/strcase"
+	"github.com/xsleonard/go-merkle"
 )
 
+// DefaultSaltsLengthSuffix is the suffix used to store the length of slices (repeated) fields in the tree. It can be
+// customized with the SaltsLenghtSuffix TreeOption
 const DefaultSaltsLengthSuffix = "Length"
 
-/*
-Allows to pass certain options to the tree generation logic. Currently support:
-	1. EnableHashSorting: Implement a merkle tree with sorted hashes
-	2. SaltsLengthSuffix: As precise proofs support repeated fields, when generating the merkle tree we need to add a leaf that represents the length of the slice.
-	The default suffix is `Length`, although it is customizable so it does not collide with potential field names of your own proto structs.
-*/
+// TreeOptions allows customizing the generation of the tree
 type TreeOptions struct {
+	//	EnableHashSorting: Implement a merkle tree with sorted hashes
 	EnableHashSorting bool
+	// SaltsLengthSuffix: As precise proofs support repeated fields, when generating the merkle tree we need to add a
+	// leaf that represents the length of the slice. The default suffix is `Length`, although it is customizable so it
+	// does not collide with potential field names of your own proto structs.
 	SaltsLengthSuffix string
+	Hash              hash.Hash
 }
 
 // DocumentTree is a helper object to create a merkleTree and proofs for fields in the document
 type DocumentTree struct {
-	propertyList      []string
 	merkleTree        merkle.Tree
+	leaves            []LeafNode
+	filled            bool
```

Comment:
Can you explain what this field does? Looks like it is a way to not generate a tree twice? But it is possible to add a document to an exiting tree?